subsys/net/openthread/rpc/server/ot_rpc_cli.c

```diff
@@ -68,32 +68,45 @@ NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_cli_init, OT_RPC_CMD_CLI_INIT, ot_
 static void ot_rpc_cmd_cli_input_line(const struct nrf_rpc_group *group,
 				      struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
 {
-	char input_line_buffer[256];
 	struct nrf_rpc_cbor_ctx rsp_ctx;
-	char *result;
+	char *buffer = NULL;
+	const void *ptr;
+	size_t len;
 
 	/* Parse the input */
-	result = nrf_rpc_decode_str(ctx, input_line_buffer, sizeof(input_line_buffer));
+	ptr = nrf_rpc_decode_str_ptr_and_len(ctx, &len);
+
+	if (ptr) {
+		buffer = malloc(len + 1);
+		if (buffer) {
+			memcpy(buffer, ptr, len);
+			buffer[len] = '\0';
+		}
+	}
 
 	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
```

Comment:
Why not move this before the allocation and copying which turn out to be unnecessary when this fails? This extra call to `free` would not be needed then.

Reply:
`nrf_rpc_decoding_done_and_check()` frees the buffer which holds the data to be stored in the `buffer`.